orcestrator-swarm/src/main/java/com/sequenceiq/cloudbreak/orcestrator/swarm/SwarmContainerOrchestrator.java

```diff
@@ -0,0 +1,283 @@
+            return new SwarmCluster(gatewayAddress, nodes, swarmManagerClient);
+        } catch (Exception e) {
+            throw new CloudbreakOrchestratorException(e);
+        }
+    }
+
+    @Override
+    public ContainerOrchestratorCluster bootstrapNewNodes(String gatewayAddress, Set<Node> nodes) throws CloudbreakOrchestratorException {
+        try {
+            DockerClient dockerApiClient = DockerClientBuilder.getInstance(getDockerClientConfig(gatewayAddress)).build();
+            Set<String> privateAddresses = getPrivateAddresses(nodes);
+            String[] cmd = {"--debug", "add", "--wait", MUNCHAUSEN_WAIT, "--join", getConsulJoinIp(gatewayAddress),
+                    prepareDockerAddressInventory(privateAddresses)};
+            new MunchausenBootstrap(dockerApiClient, MUNCHAUSEN_DOCKER_IMAGE, cmd).call();
+            DockerClient swarmManagerClient = DockerClientBuilder.getInstance(getSwarmClientConfig(gatewayAddress))
+                    .withDockerCmdExecFactory(new DockerCmdExecFactoryImpl())
+                    .build();
+            return new SwarmCluster(gatewayAddress, nodes, swarmManagerClient);
+        } catch (Exception e) {
+            throw new CloudbreakOrchestratorException(e);
+        }
+    }
+
+    @Override
+    public void startRegistrator(ContainerOrchestratorCluster cluster, String imageName) throws CloudbreakOrchestratorException {
+        try {
+            Node gateway = getGatewayNode(cluster.getApiAddress(), cluster.getNodes());
+            DockerClient swarmManagerClient = ((SwarmCluster) cluster).getDockerClient();
+            simpleContainerBootstrapRunner(new RegistratorBootstrap(swarmManagerClient, imageName, gateway.getHostname(), gateway.getPrivateIp())).call();
+        } catch (Exception e) {
+            throw new CloudbreakOrchestratorException(e);
+        }
+    }
+
+    @Override
+    public void startAmbariServer(ContainerOrchestratorCluster cluster, String dbImageName,
+            String serverImageName, String platform) throws CloudbreakOrchestratorException {
+        try {
+            DockerClient swarmManagerClient = ((SwarmCluster) cluster).getDockerClient();
+            simpleContainerBootstrapRunner(new AmbariServerDatabaseBootstrap(swarmManagerClient, dbImageName)).call();
+            simpleContainerBootstrapRunner(new AmbariServerBootstrap(swarmManagerClient, serverImageName, platform)).call();
+        } catch (Exception e) {
+            throw new CloudbreakOrchestratorException(e);
+        }
+    }
+
+    @Override
+    public void startAmbariAgents(ContainerOrchestratorCluster cluster, String imageName, int count, String platform) throws CloudbreakOrchestratorException {
+        if (count > cluster.getNodes().size()) {
+            throw new CloudbreakOrchestratorException("Cannot orchestrate more Ambari agent containers than the available nodes.");
+        }
+        try {
+            ExecutorService executor = Executors.newFixedThreadPool(TEN);
```

Comment:
Global pool would be better, since you can share the thread resources much more optimally, please consider it in future fixes.